tlsn-core/src/transcript.rs

```diff
@@ -12,6 +13,37 @@ impl Transcript {
         Self { sent, received }
     }
 
+    /// Returns a concatenated bytestring located in the given ranges of the transcript.
+    pub fn get_bytes_in_ranges(
+        &self,
+        ranges: &[TranscriptRange],
+        direction: &Direction,
+    ) -> Result<Vec<u8>, Error> {
+        // at least one range must be present
+        if ranges.is_empty() {
+            return Err(Error::InternalError);
+        }
+
+        // pick source depending on direction
+        let src = if direction == &Direction::Sent {
+            &self.sent
+        } else {
+            &self.received
+        };
+
+        let mut dst: Vec<u8> = Vec::new();
+        for r in ranges {
+            if r.end() as usize >= src.len() {
+                // range bounds must be within `src` length
+                return Err(Error::InternalError);
+            } else {
+                dst.extend(&src[r.start() as usize..r.end() as usize]);
+            }
+        }
+
+        Ok(dst)
+    }
```

Comment:
yes, it makes sense now to use std `Range`

but the rest of your comment, im not sure what you mean by

> We can achieve the same with:
> let sent_bytes = transcript.sent()[2..4].to_vec();

because this is exactly what the above code does in
`src[r.start() as usize..r.end() as usize]`

Reply:
I wanted to also perform some sanity checks on the ranges, i.e. .end > .start , .end <  .len() and !ranges.is_empty(), hence the need for this fn

but this is just a tentative fn, maybe we'll sanitize the ranges before they are passed to this fn, then we can get rid of this fn. until then i'll just keep it here.